rocketpool-daemon/api/pdao/set-signalling-address.go

```diff
@@ -108,12 +106,17 @@ func (c *protocolDaoSetSignallingAddressContext) PrepareData(data *types.TxInfoD
 		return types.ResponseStatus_Error, fmt.Errorf("Voting must be initialized to set a signalling address. Use 'rocketpool pdao initialize-voting' to initialize voting first")
 	}
 
-	sig, err := utils.ParseEIP712(c.signature)
+	signatureSanitized, err := eip712.SanitizeEIP712String(c.signature)
 	if err != nil {
-		return types.ResponseStatus_Error, fmt.Errorf("Error splitting EIP-712 signature")
+		return types.ResponseStatus_Error, fmt.Errorf("Error sanitizing signature input")
 	}
 
-	data.TxInfo, err = c.registry.SetSigner(c.signallingAddress, opts, sig.V, sig.R, sig.S)
+	components, err := eip712.ParseEIP712Components(signatureSanitized)
+	if err != nil {
+		return types.ResponseStatus_Error, fmt.Errorf("Error parsing EIP-712 components")
+	}
+
+	data.TxInfo, err = c.registry.SetSigner(c.signallingAddress, opts, components.V, components.R, components.S)
```

Comment:
we shouldn't rely on a contract revert to notify the user of an invalid signature- validate the actual signed message first. you'll have to reconstruct it.

See <https://github.com/Rocket-Rescue-Node/rescue-api/blob/main/util/ethereum.go#L22> for the broad strokes on how to recover the signer (validating the message).